sphinx/ext/apidoc/_extension.py

```diff
@@ -0,0 +1,231 @@
+"""Sphinx extension for auto-generating API documentation."""
+
+from __future__ import annotations
+
+import fnmatch
+import os
+import re
+from dataclasses import dataclass
+from typing import TYPE_CHECKING, Any
+
+from sphinx.locale import __
+from sphinx.util.console import bold
+
+from . import WARNING_TYPE, _remove_old_files, create_modules_toc_file, logger, recurse_tree
```

Comment:
Avoid relative imports:
```suggestion
from sphinx.ext.apidoc import WARNING_TYPE, _remove_old_files, create_modules_toc_file, logger, recurse_tree
```